src/graph/graph.c

```diff
@@ -420,6 +434,9 @@ int Graph_LabelTypeCount(const Graph *g) {
 void Graph_AllocateNodes(Graph *g, size_t n) {
 	assert(g);
 	DataBlock_Accommodate(g->nodes, n);
+	// Updated the required matrix size if this creation will trigger a resize.
+	size_t new_count = Graph_UncompactedNodeCount(g) + n;
+	if(new_count > Graph_RequiredMatrixDim(g)) _Graph_UpdateMatrixDimensions(g, new_count);
```

Comment:
you check this condition at `_Graph_UpdateMatrixDimensions`, so please remove one of the checks.